services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py

```diff
@@ -84,72 +148,48 @@ async def _sequentially_upsert_tasks_from_project(
-                node_details = _get_fake_service_details(service_key_version)
-            else:
-                node_details = await director_client.get_service_details(
-                    service_key_version
-                )
-                node_extras: ServiceExtras = await director_client.get_service_extras(
-                    service_key_version
-                )
-            if not node_details:
-                continue
-
-            requires_mpi = False
-            requires_gpu = False
-            if node_extras:
-                requires_gpu = NodeRequirement.GPU in node_extras.node_requirements
-                requires_mpi = NodeRequirement.MPI in node_extras.node_requirements
-            image = Image(
-                name=service_key_version.key,
-                tag=service_key_version.version,
-                requires_gpu=requires_gpu,
-                requires_mpi=requires_mpi,
-            )
+        list_of_comp_tasks_in_project = await _generate_tasks_list_from_project(
+            project, director_client, published_nodes
+        )
 
-            task_db = CompTaskAtDB(
-                project_id=project.uuid,
-                node_id=node_id,
-                schema=NodeSchema(
-                    inputs=node_details.inputs, outputs=node_details.outputs
-                ),
-                inputs=node.inputs,
-                outputs=node.outputs,
-                image=image,
-                submit=datetime.utcnow(),
-                state=(
-                    RunningState.PUBLISHED
-                    if node_id in published_nodes
-                    and node_class == NodeClass.COMPUTATIONAL
-                    else RunningState.NOT_STARTED
-                ),
-                internal_id=internal_id,
-                node_class=node_class,
+        result: RowProxy = await self.connection.execute(
+            sa.select([comp_tasks.c.node_id]).where(
+                comp_tasks.c.project_id == str(project.uuid)
             )
+        )
+        node_id_to_delete = [
+            t.node_id
+            for t in await result.fetchall()
+            if t.node_id not in project.workbench
+        ]
 
+        # start by removing the old tasks if they exist
+        for node_id in node_id_to_delete:
             await self.connection.execute(
-                insert(comp_tasks).values(
-                    **task_db.dict(by_alias=True, exclude_unset=True)
+                sa.delete(comp_tasks).where(
+                    (comp_tasks.c.project_id == str(project.uuid))
+                    & (comp_tasks.c.node_id == node_id)
                 )
             )
 
+        # insert or update the remaining tasks
+
+        for comp_task_db in list_of_comp_tasks_in_project:
+
+            insert_stmt = insert(comp_tasks).values(
+                **comp_task_db.dict(by_alias=True, exclude_unset=True)
+            )
+            on_update_stmt = insert_stmt.on_conflict_do_update(
```

Comment:
updates trigger an event in the webserver, right? Could you please add here a comment of what happens?

Reply:
update triggers only an event if it is the comp_tasks.outputs or comp_tasks.state column being updated. therefore no event is triggered here.

Reply:
added docs